linear_algebra/direct_sum_module.lean

```diff
@@ -0,0 +1,440 @@
+/-
+Copyright (c) 2018 Kenny Lau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kenny Lau
+
+Direct sum of modules over commutative rings,
+indexed by a discrete type.
+-/

```

Comment:
Isn't this `finsupp`? Or is there a difference?

Reply:
This is sort of a dependent finsupp.